tests/interlocked_int/interlocked_int.c

```diff
@@ -0,0 +1,1055 @@
+//Copyright(c) Microsoft.All rights reserved.
+//Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifdef __cplusplus
+#include <cstdlib>
+#include <cstddef>
+#include <cstdint>
+#else
+#include <stdlib.h>
+#include <stddef.h>
+#include <stdint.h>
+#endif
+
+
+#include "testrunnerswitcher.h"
+
+static TEST_MUTEX_HANDLE g_testByTest;
+
+
+
+#include "azure_c_util/interlocked.h"
+
+
+BEGIN_TEST_SUITE(interlocked_int)
+
+
+TEST_SUITE_INITIALIZE(a)
+{
+    g_testByTest = TEST_MUTEX_CREATE();
+    ASSERT_IS_NOT_NULL(g_testByTest);
+}
+
+TEST_SUITE_CLEANUP(b)
+{
+
+    TEST_MUTEX_DESTROY(g_testByTest);
+}
+
+TEST_FUNCTION_INITIALIZE(c)
+{
+    if (TEST_MUTEX_ACQUIRE(g_testByTest))
+    {
+        ASSERT_FAIL("our mutex is ABANDONED. Failure in test framework");
+    }
+}
+
+TEST_FUNCTION_CLEANUP(d)
+{
+    TEST_MUTEX_RELEASE(g_testByTest);
+}
+
+/*Tests_SRS_INTERLOCKED_43_001: [ interlocked_add shall atomically add 32-bit integers *addend and value and store the result in *addend.]*/
+/*Tests_SRS_INTERLOCKED_43_032: [interlocked_add shall return the result of the addition.]*/
+TEST_FUNCTION(interlocked_add_does_addition)
+{
+    ///arrange
+    volatile int32_t addend;
+	interlocked_exchange(&addend, INT32_MAX);
+    int32_t value = INT32_MIN;
+
+    ///act
+    int32_t return_val = interlocked_add(&addend, value);
+
+    ///assert
+    ASSERT_ARE_EQUAL(int32_t, -1, interlocked_or(&addend, 0));
```

Comment:
> 0)); [](start = 57, length = 5)

Consider adding some information of what failed in the assert (ASSERT_ARE_EQUAL allows you to say ASSERT_ARE_EQUAL(int32_t, -1, interlocked_or(&addend, 0), "addend was not modified");
This makes sense if there are multiple asserts in a test and they are of the same type or same style. Having some extra bit of text (besides the expected value, actual value and line no makes reading the result of the test a bit easier).
Not mandatory, but a consider.